### PR TITLE
[WIP] Initial work on ordered_use_length fixer

### PR DESCRIPTION
**Pull request by GrahamCampbell**
This is currently broken and the test cases are totally wrong. This is a wip...

### CODE REVIEW

File: Symfony/CS/AbstractOrderedUseFixer.php [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+            $uses = array_reverse($uses);
+            $usesOrder = array_replace($usesOrder, self::getNewOrder($uses, $tokens));
+        }
+
+        // First clean the old content
+        // This must be done first as the indexes can be scattered
+        foreach ($usesOrder as $use) {
+            for ($i = $use[1]; $i <= $use[2]; ++$i) {
+                $tokens[$i]->clear();
+            }
```
Comment (stof) [inline Symfony/CS/AbstractOrderedUseFixer.php]: use ``$tokens->clearRange($use[1], $use[2]);`` instead of reimplementing the same

Comment (GrahamCampbell) [inline Symfony/CS/AbstractOrderedUseFixer.php]: @stof I haven't actually changed this code. It's identical to what it was when stof wrote it. Maybe you could send a pull to the current implementation to change it?

Comment (keradus) [inline Symfony/CS/AbstractOrderedUseFixer.php]: done: #1049

**Comment by keradus** [comment]
out of curiosity... any framework use that kind of import sorting ?

**Comment by GrahamCampbell** [comment]
> out of curiosity... any framework use that kind of import sorting ?

Yep, laravel.

**Comment by GrahamCampbell** [comment]
We're moving to PSR-2 for laravel 5.1, but we also need some other fixers, like this one :)

**Comment by GrahamCampbell** [comment]
I will start again on this in July when I have way, way more time.

**Comment by keradus** [comment]
Or just consider using current ordered_use fixer ;)
It will need less time to do so ;)

**Comment by GrahamCampbell** [comment]
@keradus If it were up to me, we would, but Taylor wants ordered by length...

**Comment by keradus** [comment]
OK, I see.
It is always great to have sth standarized and automated ;)

**Pull request closed without merging**